core/domain/acl_decorators.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Decorators to provide authorization across the site."""
+
+from core.domain import rights_manager
+from core.domain import role_services
+import feconf
+
+
+def check_activity_accessible(self, activity_id, activity_type):
+    """Returns a boolean to signify whether given activity is accessible
+    by the user or not.
+
+    Args:
+        activity_id: str. Id of the given activity.
+        activity_type: str. Signifies whether activity is exploration or
+            collection.
+
+    returns:
+        bool. True if activity is accessible else False.
+    """
+    if activity_type == feconf.ACTIVITY_TYPE_EXPLORATION:
```

Comment:
On further investigation, let's just get rid of the special handling of this altogether. It's fine to just serve a usual 404 (so just return False here). We only have one disabled exploration and it's been disabled for more than a year by now, so I don't think we need a special page for it.